windows-agent/internal/proservices/landscape/landscape_test.go

```diff
@@ -828,13 +828,19 @@ func TestNotifyConfigUpdate(t *testing.T) {
 	testcases := map[string]struct {
 		emptyDB   bool
 		conf, uid string
+
+		want        string
+		wantNoTasks bool
 	}{
 		"Task contains client conf when UID is not empty":                                       {},
 		"Task contains empty client conf":                                                       {conf: "-"},
 		"Task contains empty client conf when UID is empty despite submitted conf is not empty": {uid: "-"},
 		"Task contains empty client conf when both are empty":                                   {conf: "-", uid: "-"},
+		"Task doesn't contain [host] section":                                                   {conf: "[host]\nurl=localhost\n[client]\nurl=another\n", want: "[client]\nurl=another\n"},
```

Comment:
So, we are filtering and testing the task itself here. What about `agent.yaml` then? Is there corresponding tests to ensure the one we deliver doesn’t have an `[host]` section too, or is this one delivered via `NotifyConfigUpdate` too?

Reply:
The way the cloudinit package serializes agent.yaml requires it to know about the structure of the INI configuration, it already does the filtering by itself:

```go
f, err := ini.Load(strings.NewReader(conf))
	if err != nil {
		return fmt.Errorf("could not load Landscape configuration file")
	}

	section, err := f.GetSection("client")
	if err != nil {
		return nil // Empty section
	}

	landscapeModule.Client = make(map[string]string)
	for _, keyName := range section.KeyStrings() {
		landscapeModule.Client[keyName] = section.Key(keyName).String()
	}

	out["landscape"] = landscapeModule
	return nil
```

That's why I didn't touch it.

Reply:
Sorry, what I meant is that is this tested through those tests?
Like, we have a file that is filtered and ensure we end up in agent.yaml without the host section? (even if we know it doesn’t touch at all the file, we document the behaviour that way)

Reply:
I created a different test function for that to avoid loosing coverage on some internal error paths that would otherwise be filtered by the `config` methods.

Reply:
yeah, I just saw that, it’s better that way indeed! :)